torch_xla/csrc/tensor_impl.cpp

```diff
@@ -1,15 +1,69 @@
 #include "tensor_impl.h"
 
+#include <c10/core/impl/DeviceGuardImplInterface.h>
+#include <c10/macros/Macros.h>
+
 #include "tensor_util.h"
 #include "tensorflow/compiler/xla/xla_client/debug_macros.h"
 
 namespace torch_xla {
+namespace {
+
+struct XLAGuardImpl : public c10::impl::DeviceGuardImplInterface {
+  at::DeviceType type() const override { return at::DeviceType::XLA; }
+
+  c10::Device exchangeDevice(c10::Device) const override {
+    return c10::Device(at::DeviceType::XLA, 0);
+  }
+
+  c10::Device getDevice() const override {
+    return c10::Device(at::DeviceType::XLA, 0);
+  }
+
+  void setDevice(c10::Device) const override {}
+
+  void uncheckedSetDevice(c10::Device d) const noexcept override {}
+
+  c10::Stream getStream(c10::Device d) const noexcept override {
+    return c10::Stream(c10::Stream::DEFAULT,
+                       c10::Device(at::DeviceType::XLA, 0));
+  }
+
+  c10::Stream exchangeStream(c10::Stream s) const noexcept override {
+    return c10::Stream(c10::Stream::DEFAULT,
+                       c10::Device(at::DeviceType::XLA, 0));
+  }
+
+  c10::DeviceIndex deviceCount() const override { return 1; }
+};
+
+}  // namespace
+
+C10_REGISTER_GUARD_IMPL(XLA, XLAGuardImpl);
 
 // TODO: Replace UndefinedTensorId with proper type.
```

Comment:
Remove this TODO.

Reply:
Can we do this is a separate PR?
Right now I am confused about this merging of the outstanding 347 commits we have accumulated.

Reply:
Haha ok, I don't see what's so scary about an additional one liner commit.